torch/_dynamo/variables/misc.py

```diff
@@ -790,6 +791,14 @@ def wraps(fn):
             return variables.lists.DequeVariable(
                 items, mutable_local=MutableLocal(), **options
             )
+        elif self.value is functools.partial:
+            guards = options.get("guards", set())
+            guards.add(self.source.make_guard(GuardBuilder.TYPE_MATCH))
```

Comment:
Is this kosher? Won't you unpredictably mutate options["guards"] that was passed in as an argument?

Reply:
This is definitely not enough guards.  The correct way to handle this is to recrusively guard on:
- value.func
- value.args
- value.keywords

Reply:
Yeah we should probably use regular old propagation, good catch